zcash_primitives/src/legacy/keys.rs

```diff
@@ -0,0 +1,198 @@
+    /// `m/44'/<coin_type>'/<account>'
+    ///
+    /// This produces the extended private key for the external (non-change)
+    /// address at the specified index for the provided account.
+    pub fn from_seed<P: consensus::Parameters>(
+        params: &P,
+        seed: &[u8],
+        account: AccountId,
+    ) -> Result<AccountPrivKey, hdwallet::error::Error> {
+        ExtendedPrivKey::with_seed(&seed)?
+            .derive_private_key(KeyIndex::hardened_from_normalize_index(44)?)?
+            .derive_private_key(KeyIndex::hardened_from_normalize_index(params.coin_type())?)?
+            .derive_private_key(KeyIndex::hardened_from_normalize_index(account.0)?)
+            .map(AccountPrivKey)
+    }
+
+    pub fn from_extended_privkey(extprivkey: ExtendedPrivKey) -> Self {
+        AccountPrivKey(extprivkey)
+    }
+
+    pub fn extended_privkey(&self) -> &ExtendedPrivKey {
+        &self.0
+    }
+
+    pub fn to_account_pubkey(&self) -> AccountPubKey {
+        AccountPubKey(ExtendedPubKey::from_private_key(&self.0))
+    }
+
+    /// Derive BIP-44 private key at the external child path
+    /// `m/44'/<coin_type>'/<account>'/0/<child_index>
+    pub fn derive_external_secret_key(
+        &self,
+        child_index: u32,
+    ) -> Result<ExternalPrivKey, hdwallet::error::Error> {
+        self.0
+            .derive_private_key(KeyIndex::Normal(0))?
+            .derive_private_key(KeyIndex::Normal(child_index))
+            .map(ExternalPrivKey)
+    }
+}
+
+/// A type representing a BIP-44 public key at the account path level
+/// `m/44'/<coin_type>'/<account>'
+#[derive(Clone, Debug)]
+pub struct AccountPubKey(ExtendedPubKey);
+
+impl AccountPubKey {
+    /// Derive BIP-44 public key at the external child path
+    /// `m/44'/<coin_type>'/<account>'/0/<child_index>
+    pub fn derive_external_pubkey(
+        &self,
+        child_index: u32,
+    ) -> Result<ExternalPubKey, hdwallet::error::Error> {
+        self.0
+            .derive_public_key(KeyIndex::Normal(0))?
+            .derive_public_key(KeyIndex::Normal(child_index))
+            .map(ExternalPubKey)
+    }
+
+    pub fn from_extended_pubkey(extpubkey: ExtendedPubKey) -> Self {
+        AccountPubKey(extpubkey)
+    }
+
+    pub fn extended_pubkey(&self) -> &ExtendedPubKey {
+        &self.0
+    }
+}
+
+/// A type representing a private key at the BIP-44 external child
+/// level `m/44'/<coin_type>'/<account>'/0/<child_index>
+#[derive(Clone, Debug)]
+pub struct ExternalPrivKey(ExtendedPrivKey);
+
+impl ExternalPrivKey {
+    /// Returns the external public key corresponding to this private key
+    pub fn to_external_pubkey(&self) -> ExternalPubKey {
+        ExternalPubKey(ExtendedPubKey::from_private_key(&self.0))
+    }
+
+    /// Extracts the secp256k1 secret key component
+    pub fn secret_key(&self) -> &secp256k1::key::SecretKey {
+        &self.0.private_key
+    }
+}
+
+pub fn pubkey_to_address(pubkey: &secp256k1::key::PublicKey) -> TransparentAddress {
```

Comment:
We can use the [`#[deprecated]` attribute](https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-deprecated-attribute) to indicate that this is going away.
```suggestion
/// Derives the P2PKH transparent address corresponding to the given pubkey.
#[deprecated(note = "This function will be removed in an upcoming refactor.")]
pub fn pubkey_to_address(pubkey: &secp256k1::key::PublicKey) -> TransparentAddress {
```